packages/effect/src/internal/channel.ts

```diff
@@ -1217,7 +1217,7 @@ export const mergeAllWith = (
                 )
             })
           ),
-          Effect.repeatUntil(Option.isSome),
+          Effect.repeat({ until: (_): _ is Option.Some<OutDone> => Option.isSome(_) }),
```

Comment:
This is a case where tacit usage of refinements can no longer be applied, but maybe this ok.